crates/api/src/site/tagline.rs

```diff
@@ -0,0 +1,59 @@
+use crate::Perform;
+use actix_web::web::Data;
+use lemmy_api_common::{
+  tagline::{GetRandomTagline, GetRandomTaglineResponse, ListTaglines, ListTaglinesResponse},
+  utils::{blocking, get_local_user_view_from_jwt_opt, is_admin},
+};
+use lemmy_db_schema::source::tagline::Tagline;
+use lemmy_utils::{error::LemmyError, ConnectionId};
+use lemmy_websocket::LemmyContext;
+
+#[async_trait::async_trait(?Send)]
+impl Perform for ListTaglines {
+  type Response = ListTaglinesResponse;
+
+  #[tracing::instrument(skip(context, _websocket_id))]
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    _websocket_id: Option<ConnectionId>,
+  ) -> Result<ListTaglinesResponse, LemmyError> {
+    let data: &ListTaglines = self;
+
+    let local_user_view =
+      get_local_user_view_from_jwt_opt(data.auth.as_ref(), context.pool(), context.secret())
+        .await?;
+    let is_admin = match local_user_view {
+      Some(s) => is_admin(&s).is_ok(),
+      None => false,
+    };
+
+    // Only let admins read this
+    if !is_admin {
+      return Err(LemmyError::from_message("not_an_admin"));
+    }
+
+    let taglines = blocking(context.pool(), Tagline::get_all).await??;
+
+    Ok(ListTaglinesResponse { taglines })
+  }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Perform for GetRandomTagline {
+  type Response = GetRandomTaglineResponse;
+
+  #[tracing::instrument(skip(context, _websocket_id))]
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    _websocket_id: Option<ConnectionId>,
+  ) -> Result<GetRandomTaglineResponse, LemmyError> {
+    let content = match blocking(context.pool(), Tagline::get_random).await? {
+      Some(s) => s.content,
+      None => "".to_string(),
```

Comment:
Have GetSiteResponse include `tagline: Option<String>`. Then you won't have to do anything but `.ok()` on the get_random result. 

The front ends will know that this field is optional, and only render it if it exists.